Core/clim-core/presentation-translators.lisp

```diff
@@ -334,26 +319,26 @@ and used to ensure that presentation-translators-caches are up to date.")
 
 (defun find-presentation-translators (from-type to-type command-table)
   (let* ((command-table (find-command-table command-table))
-         (from-name (presentation-type-name from-type))
-         (to-name (presentation-type-name to-type))
-         (cache-key (cons from-name to-name))
-         (cache-table (presentation-translators-cache
-                       (presentation-translators command-table)))
-         (cached-translators (gethash cache-key cache-table)))
-    (when cached-translators
+         (from-name     (presentation-type-name from-type))
+         (to-name       (presentation-type-name to-type))
+         (cache-key     (cons from-type to-name))
+         (cache-table   (presentation-translators-cache
+                         (presentation-translators command-table))))
+    (when-let ((cached-translators (gethash cache-key cache-table)))
       (return-from find-presentation-translators cached-translators))
     (let ((translator-vector (make-array 8 :adjustable t :fill-pointer 0))
           (table-counter 0))
       (do-command-table-inheritance (table command-table)
-        (let ((translator-map (simple-type-translators
-                               (presentation-translators table))))
-          (flet ((get-translators (super)
-                   (loop
-                     for translator in (gethash (type-name super) translator-map)
-                     if (stupid-subtypep (to-type translator) to-type)
-                       do (vector-push-extend (cons translator table-counter)
-                                              translator-vector))))
-            (map-over-ptype-superclasses #'get-translators from-name)))
+        ;; We need to go over each translator in all tables, because
+        ;; the FROM-TYPE may be a subtype of a translator FROM-TYPE
+        ;; despite not having the same presentation metaclass, like
+        ;; "meta" presentation types OR, AND etc. -- jd 2020-06-24
+        (loop with translators = (translators (presentation-translators table))
+              for tr being the hash-value of translators
```

Comment:
Is the hash-table still used as a hash-table somewhere? Otherwise changing the representation could make sense.

Reply:
it is used in many places, add-translator, and remove-translator in this file, plus in commands.lisp (i.e find-presentation-translator)

Reply:
OK.